src/runtimes/constants.ts

```diff
@@ -0,0 +1 @@
+export const FUNCTIONS_INTERNAL_DIR = '.netlify/internal-functions'
```

Comment:
There's something about this that doesn't feel right to me. When we added support for the internal functions directory, we made the decision to keep that abstraction outside of zip-it-and-ship-it, such that [it only receives a list of directories](https://github.com/netlify/zip-it-and-ship-it/blob/main/src/zip.ts#L50) without knowing anything about them (i.e. which ones contain user-generated or framework-generated functions).

With this change, we're making zip-it-and-ship-it aware of the concept of internal functions, but only partially. On the one hand, we're giving it a list of directories without conveying the type, and on the other hand we're expecting zip-it-and-ship-it to know what the internal directory is.

In my mind, it would make more sense to add a parameter specifically for the internal functions directory. If found, it's added to the list of directories to be traversed and it will also be used to populate the `isInternalFunction` property.

Does that make sense?

Reply:
Yeah it makes sense :) I'll start with Daniels suggestions, and then implement this!